mono/metadata/custom-attrs.c

```diff
@@ -1618,20 +1618,21 @@ mono_custom_attrs_from_index_checked (MonoImage *image, guint32 idx, gboolean ig
 	if (!i)
 		return NULL;
 	i --;
+	attr_array = g_array_new (TRUE, TRUE, sizeof (guint32));
```

Comment:
Can you presize this to reduce heap allocations?
For example, if it was a std::vector, you might say something like `.resize(100)` or `.resize(1000)`.
Also, if this function is called more than once for the same inputs (is it?) can you memorize a size here? Or maybe all the data?

Reply:
I can presize it yes, `g_array_sized_new` will do that. Unfortunately I couldn't find a good way to predict the size in practice.

Reply:
In STL you give a number to presize, but you aren't bound do it.
There is allocated size ("capacity") and used size ("size").
You can ask for an allocated size, and then if you add past that, it grows for you.
If you don't add to it, it is wasteful, but not hugely, depending.
There is going to be a heap allocation either way, you can at least inflate the first one by some heuristic.

Reply:
Yeah, it's the same case with `g_array_sized_new`. As I said though, I don't have a good way to anticipate the size on this - I guess I could allocate a couple hundred and call it good since it's only temporary. Probably better than the initial allocation value (16...).